### PR TITLE
Fix unsafe json construction for gce related classes

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

 /kind bug

**What this PR does / why we need it**:

Fix unsafe json construction.

**Which issue(s) this PR fixes**:

Xref #81134

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
none
```

### CODE REVIEW

**Comment by gavinfish** [comment]
/assign @cheftako
/cc @liggitt
I think there no need to change the test codes for this issue, right? @liggitt

**Comment by smourapina** [comment]
Bug Triage for release 1.18. Adding this to milestone 1.18, as this is one of the PRs linked to [#81134](https://github.com/kubernetes/kubernetes/issues/81134)
Please have in mind that we are 2 days away from code freeze (5 March)!
/milestone v1.18

**Comment by bai** [comment]
👋 Hello from Bug Triage team! This PR has not been updated for a while, so I'd like to check its status. The code freeze is starting June 25th (about 4 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

As the PR is tagged for 1.19, is it still planned for this release?

**Comment by bai** [comment]
As per conversation on Slack, this issue is not going to make it into 1.19.

/milestone v1.20

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_external.go [hunk 1]
```diff
@@ -554,9 +554,16 @@ func (g *Cloud) createTargetPoolAndHealthCheck(svc *v1.Service, name, serviceNam
 		instances = append(instances, host.makeComparableHostPath())
 	}
 	klog.Infof("Creating targetpool %v with %d healthchecks", name, len(hcLinks))
+	description := forwardingRuleDescription{
+		ServiceName: serviceName,
+	}
+	descriptionStr, err := description.marshal()
+	if err != nil {
+		return err
+	}
```
Comment (cheftako) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_external.go:563]: Can we add simple test for this block of code? Essentially demonstrating that we are not changing the generated description string.

**Comment by cheftako** [comment]
@gavinfish Are you still working on this?

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_external.go (hunk 1 above)
Comment (gavinfish) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_external.go:560]: added.

**Comment by gavinfish** [comment]
> @gavinfish Are you still working on this?

@cheftako sorry for the delay. I have updated the PR. PTAL, thanks.

**Comment by gavinfish** [comment]
@cheftako PTAL

**Comment by dims** [comment]
@cheftako this has been struggling to land since Mar 1, 2020, PTAL

/retest

**Pull request closed without merging**